core/src/main/java/com/google/cloud/sql/Connector.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.sql;
+
+/** Manage named connector configurations, leveraging the JDBC URL. */
+public interface Connector {
```

Comment:
What is the purpose of this interface? Do we need an interface when only one class implements it? Also why is it public? It's not part of our public API itself.

Reply:
Let's discuss.